util/feature_detection.go

```diff
@@ -4,7 +4,25 @@ import (
 	"runtime"
 )
 
-// Does the runtime OS support docker natively, versus needing to run docker in a virtual machine
+// Constants for virtualization drivers
+const (
+	Xhyve      = "xhyve"
+	VMWare     = "vmwarefusion"
+	VirtualBox = "virtualbox"
+)
+
+// SupportsNativeDocker determines if the runtime OS support docker natively,
+// versus needing to run docker in a virtual machine
 func SupportsNativeDocker() bool {
 	return runtime.GOOS == "linux"
 }
+
+// IsMac detects if we are running on the darwin platform
```

Comment:
Shouldn't we have IsLinux?
Making the other checks more feature driven seems a fine followup

Reply:
Are you sugesting an IsLinux in addition to SupportsNativeDocker()?  Most of the linux checks are for a feature (native docker support) where as some of the mac checks were for things like which route command to use or this process to kill to restart dns resolution. So there is a bit of the mix in platform name and feature detection.  This is the quicked line to remove linter warnings and also make the code more readable (and centralized if for some crazy reason the string "darwin" goes away.

Reply:
My thinking was for API completeness if we have checks for the OS's we may as well have one for each that we support.